frontend/src/upload.js

```diff
@@ -245,76 +240,81 @@ $(document).ready(function() {
       cd5: window.referrer
     });
 
-    fileSender
-      .upload()
-      .then(info => {
-        const endTime = Date.now();
-        const totalTime = endTime - startTime;
-        const uploadTime = endTime - uploadStart;
-        const uploadSpeed = file.size / (uploadTime / 1000);
-
-        // record upload-stopped (completed) by sender
-        sendEvent('sender', 'upload-stopped', {
-          cm1: file.size,
-          cm2: totalTime,
-          cm3: uploadSpeed,
-          cm5: storage.totalUploads,
-          cm6: unexpiredFiles,
-          cm7: storage.totalDownloads,
-          cd1: event.type === 'drop' ? 'drop' : 'click',
-          cd2: 'completed'
-        });
-
-        const fileData = {
-          name: file.name,
-          size: file.size,
-          fileId: info.fileId,
-          url: info.url,
-          secretKey: info.secretKey,
-          deleteToken: info.deleteToken,
-          creationDate: new Date(),
-          expiry: expiration,
-          totalTime: totalTime,
-          typeOfUpload: event.type === 'drop' ? 'drop' : 'click',
-          uploadSpeed: uploadSpeed
-        };
-
-        storage.addFile(info.fileId, fileData);
-        $('#upload-filename').attr(
-          'data-l10n-id',
-          'uploadSuccessConfirmHeader'
-        );
-        t = window.setTimeout(() => {
+    // For large files we need to give the ui a tick to breathe and update
+    // before we kick off the FileSender
+    setTimeout(() => {
+      fileSender
+        .upload()
+        .then(info => {
+          const endTime = Date.now();
+          const totalTime = endTime - startTime;
+          const uploadTime = endTime - uploadStart;
+          const uploadSpeed = file.size / (uploadTime / 1000);
+          const expiration = 24 * 60 * 60 * 1000; //will eventually come from a field
```

Comment:
Unrelated, but not sure if this can be replaced with the `ONE_DAY_IN_MS` const imported above.